src/Intercom/Clients/AdminsClient.cs

```diff
@@ -64,12 +64,12 @@ public Admin View (String id)
         public Admin View(Admin admin)
         {
             if (admin == null) {
-                throw new ArgumentNullException ("'admin' argument is null.");
+                throw new ArgumentNullException (nameof(admin));
             }
 
             if (String.IsNullOrEmpty(admin.id))
             {
-                throw new ArgumentNullException ("you must provide value for 'admin.id'.");
+                throw new ArgumentException ("you must provide value for 'admin.id'.");
```

Comment:
Just wondering, shouldn't this be `ArgumentNullException` since the id could be `null or empty`?